src/librustdoc/html/static/js/search.js

```diff
@@ -154,13 +389,101 @@ window.initSearch = function(rawSearchIndex) {
             return NO_TYPE_FILTER;
         }
 
-        var valLower = query.query.toLowerCase(),
-            val = valLower,
-            typeFilter = itemTypeFromName(query.type),
-            results = {}, results_in_args = {}, results_returned = {},
-            split = valLower.split("::");
+        userQuery = userQuery.trim();
+        var query = {
+            original: userQuery,
+            userQuery: userQuery.toLowerCase(),
```

Comment:
Thanks for renaming `val` to `userQuery`. It's much clearer. But that made me wonder "what's the difference between `original` and `userQuery`?" It looks like the only difference is that `userQuery` is lowercased, right?

It seems that `original` is only needed for display purposes, by the caller of `parseQuery`. It's not needed as an output of `parseQuery`. Can we get rid of it?

Reply:
We need it for aliases as well. I'm not sure it'd be a good idea to remove it like this.

Reply:
Because they might be very "broken" from a parser point of view. For example `<=` is a valid alias. So we simply use the original query to check if any doc alias matches it (it only has literal checks on doc aliases).